### DISCUSSION TITLE
Should refresh() method load relations declared in $with variable?

### DISCUSSION THREAD

**Discussion by alexandrubau** [laravel/framework#40708] [Ideas] [answered]
I've recently started working on a new Laravel application and I came across the following situation:

```php
class User extends Model
{
    protected $with = ['profile'];

    public function profile(): HasOne
    {
        return $this->hasOne(UserProfile::class);
    }
}
```

```php
class UserProfile extends Model
{
    protected $touches = ['user'];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
```

```php
class NewController extends Controller {
    public function store()
    {
         // .. 
        $user = new User($userData);
        $profile = new UserProfile($profileData);
        
        $user->save();
        $user->profile()->save($profile);
        
        $user->refresh(); // <-- Here the "profile" relation is not going to be loaded, so $user->relationLoaded('profile') returns false
        // $user->load('profile') // <-- You have to load the specific relation or do User::find($user->id); 
        
        return response()->json(['data' => $user], 201);
    }
}
```

Upon calling the `$user->refresh()` model I was expecting to include also the relations that were declared inside the `$with` variable. Basically I was expecting `$user->refresh()` to behave more like `User:find(<id>)`.

I know that in [Laravel's Docs](https://laravel.com/docs/8.x/eloquent-relationships#the-save-method) the behaviour of the `refresh()` method is correctly described but why is `$user->refresh()` limited to loading only the previously loaded relations?

What do you think about altering the behaviour of the refresh method?

**Comment by gguney** [comment, marked as answer]
You are not loading the relation with calling `$user->profile()->save($profile);` if you look at the Model.php inside`Illuminate\Database\Eloquent` it **is** loading the relations.
